subt_ign/src/GameLogicPlugin.cc

```diff
@@ -1012,6 +1049,174 @@ void GameLogicPluginPrivate::OnEvent(const ignition::msgs::Pose &_msg)
+            if (this->dataPtr->keInfo.find(model->Data()) !=
+                this->dataPtr->keInfo.end()) {
+              return true;
+            }
+
+            std::vector<Entity> links = _ecm.EntitiesByComponents(
+                components::ParentEntity(model->Data()), components::Link());
+
+            // Get the mass for the robot by summing the mass of each link,
+            // and use the canonical link for KE computation.
+            double mass = 0.0;
+            for (const Entity &link : links)
+            {
+              auto inertial = _ecm.Component<components::Inertial>(link);
+              auto canonical =  _ecm.Component<components::CanonicalLink>(link);
+              mass += inertial->Data().MassMatrix().Mass();
+
+              if (canonical)
+              {
+                this->dataPtr->keInfo[model->Data()].link += link;
+                // Create a world pose component if one is not present.
+                if (!_ecm.Component<components::WorldPose>(link))
+                {
+                  _ecm.CreateComponent(link, components::WorldPose());
+                }
+
+                // Create an inertial component if one is not present.
+                if (!_ecm.Component<components::Inertial>(link))
+                {
+                  _ecm.CreateComponent(link, components::Inertial());
+                }
+
+                // Create a world linear velocity component if one is not
+                // present.
+                if (!_ecm.Component<components::WorldLinearVelocity>(link))
+                {
+                  _ecm.CreateComponent(link,
+                      components::WorldLinearVelocity());
+                }
+
+                // Create an angular velocity component if one is not present.
+                if (!_ecm.Component<components::AngularVelocity>(link))
+                {
+                  _ecm.CreateComponent(link,
+                      components::AngularVelocity());
+                }
+
+                // Create a world angular velocity component if one is not
+                // present.
+                if (!_ecm.Component<components::WorldAngularVelocity>(
+                      link))
+                {
+                  _ecm.CreateComponent(link,
+                      components::WorldAngularVelocity());
+                }
+              }
+            }
+
+            if (mass > 0)
+            {
+              // This sets the kinetic energy threshold for the robot. It is
+              // based on the kinetic energy formula KE = 0.5 * M * v * v
+              //
+              // M is the mass of the robot.
+              // v is velocity of the robot. We are using the velocity acheived
+              // by falling from a height.
+              this->dataPtr->keInfo[model->Data()].kineticEnergyThreshold =
+                0.5 * mass * std::pow(
+                    sqrt((2 * this->dataPtr->keHeight) / 9.8) * 9.8, 2);
```

Comment:
This ends up evaluating equivocally however the velocity can be simplified from:
`sqrt((2 * this->dataPtr->keHeight) / 9.8) * 9.8`
to:
`sqrt((2 * this->dataPtr->keHeight) * 9.8)`
which would omit the cost of one division.

Reply:
Done in 102aa622dc0ad1b5eeef35903dc42774ed889f7c